### PR TITLE
Bluetooth: Fix Wrong format when debug

### PR DESCRIPTION
**Pull request by LingaoM**
Fix wformat build warning

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -3202,7 +3202,7 @@ static void hci_vs_init(void)
 		vs_hw_variant(sys_le16_to_cpu(rp.info->hw_platform),
 			      sys_le16_to_cpu(rp.info->hw_variant)),
 		sys_le16_to_cpu(rp.info->hw_variant));
-	BT_INFO("Firmware: %s (0x%02x) Version %u.%u Build %u",
+	BT_INFO("Firmware: %s (0x%02x) Version %u.%u Build %lu",
```
Comment (Thalley) [inline subsys/bluetooth/host/hci_core.c:3205]: Shouldn't `%u` be fine to use for `uint32_t` values?

File: subsys/bluetooth/host/smp.c [hunk 2]
```diff
@@ -710,7 +710,7 @@ static int smp_g2(const uint8_t u[32], const uint8_t v[32],
 	memcpy(passkey, xs + 12, 4);
 	*passkey = sys_be32_to_cpu(*passkey) % 1000000;
 
-	BT_DBG("passkey %u", *passkey);
+	BT_DBG("passkey %d", *passkey);
```
Comment (Thalley) [inline subsys/bluetooth/host/smp.c:713]: `*passkey` is a `uint32_t` and should be `%u`.

File: subsys/bluetooth/host/smp.c [hunk 3]
```diff
@@ -2375,7 +2375,7 @@ static uint8_t legacy_request_tk(struct bt_smp *smp)
 		}
 
 		if (IS_ENABLED(CONFIG_BT_LOG_SNIFFER_INFO)) {
-			BT_INFO("Legacy passkey %u", passkey);
+			BT_INFO("Legacy passkey %d", passkey);
```
Comment (Thalley) [inline subsys/bluetooth/host/smp.c:2378]: `*passkey` is a `uint32_t` and should be `%u`.

**Review by Thalley — DISMISSED** [review]

**Review by trond-snekvik — APPROVED** [review]
Mesh part looks good to me

File: subsys/bluetooth/mesh/gatt_services.c [hunk 4]
```diff
@@ -1107,7 +1107,7 @@ static size_t gatt_prov_adv_create(struct bt_data prov_sd[1])
 
 	prov_sd[0].type = BT_DATA_URI;
 	prov_sd[0].data_len = uri_len;
-	prov_sd[0].data = prov->uri;
+	prov_sd[0].data = (const uint8_t *)prov->uri;
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/gatt_services.c:1110]: This doesn't seem to be debug/log string related, so please split to another commit.

File: subsys/bluetooth/mesh/cfg_cli.c [hunk 5]
```diff
@@ -1975,7 +1975,7 @@ static int mod_sub_va(uint32_t op, uint16_t net_idx, uint16_t addr, uint16_t ele
 	}
 
 	BT_DBG("net_idx 0x%04x addr 0x%04x elem_addr 0x%04x label %s",
-	       net_idx, addr, elem_addr, log_strdup(label));
+	       net_idx, addr, elem_addr, log_strdup((const char *)label));
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/cfg_cli.c:1978]: Is `label` even even null-terminated? I don't think so - it's just a 128-bit "UUID", right @trond-snekvik ? So there's a bug in the original code that needs fixing and doing the type cast just hides the bug even more.

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/cfg_cli.c (hunk 5 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/cfg_cli.c:1978]: Good point, yes, it doesn't have to be.

Comment (LingaoM) [reply subsys/bluetooth/mesh/cfg_cli.c:1978]: 👍

Comment (LingaoM) [reply subsys/bluetooth/mesh/cfg_cli.c:1978]: I'm surprised that it's always there.

**Review by jhedberg — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]
